### PR TITLE
Hide scrollbars when nothing to scroll

### PR DESCRIPTION
**Pull request by magroader**
This creates a new utility script to help with putting vertical lists and scrollbars together. It simplifies the code of using the two of them significantly, while also hiding the scrollbar thumb when it's not available.

Currently works for Natures list and Moves list.

The moves list had a particular GUI setup that was not shared by the Natures list or, indeed, MOST lists. I plan to update the script to allow you to specify the structure in advance, for example:

```
self.my_list = gooey_scrolling_list.create("list_id", "scroll_area", "list_item", "scrollbar_handle", "scrollbar_bar", callback, other_callback, etc.)
...
self.my_list.on_input(action_id, action)
```

#107

### CODE REVIEW

**Comment by magroader** [comment]
Every scrollbar in the app now uses a consistent utility. This ended up culling a lot of repetitive code (notably, all the places the node string for scrollbars / lists were repeated in the same file) and making all the scrollbars behave consistently.

Scrollbar handles now size appropriately to the size of the scroll area.
Scrollbar handles now disappear when not needed (though the scroll background still stays).

File: screens/generate_pokemon/dropdown.lua [hunk 1]
```diff
@@ -1,135 +1,159 @@
-local gooey = require "screens.generate_pokemon.gooey.gooey"
```
Comment (magroader) [inline screens/generate_pokemon/dropdown.lua:1]: I don't quite understand what was going on here. There were 3 excess files under generate_pokemon with many unused functions, all of them only used by this dropdown. They also didn't seem to provide any value at all. I deleted all of them and just used gooey.gooey.

File: screens/generate_pokemon/dropdown.lua [hunk 2]
```diff
@@ -1,135 +1,159 @@
...
+local gooey = require "gooey.gooey"
 local gooey_buttons = require "utils.gooey_buttons"
+local gooey_scrolling_list = require "utils.gooey_scrolling_list"
 
 local M = {}
 
 local active = {active=false}
 
-local function button_click(name, button_id, scroll_id)
+local function button_click(data)
```
Comment (magroader) [inline screens/generate_pokemon/dropdown.lua:9]: This file was the most changed of everything - I turned dropdown into an object you can "create" and call functions on. All of the data associated with a single dropdown (of which there are several in generate_pokemon) is carried around in this "data" object.

File: screens/pokedex/pokedex.gui_script [hunk 3]
```diff
@@ -79,7 +80,7 @@ local function update_dex_data(self)
 	self.pokemon_rows = table_of_rows(self)
 end
 
-local function update_items(item)
+local function update_item(list, item)
```
Comment (magroader) [inline screens/pokedex/pokedex.gui_script:83]: When I found "update_items" I changed it to "update_item" to be less confusing - from the list's perspective, there is only 1 item here.

File: screens/pokedex/pokedex.gui_script [hunk 4]
```diff
@@ -153,6 +144,27 @@ local function update_pokedex_info(self)
 	gui.set_text(gui.get_node("txt_caught_kalos"), dex.get_region_seen(dex_data.regions.KALOS))
 end
 
+local function scroll_list_click(self, list)
```
Comment (magroader) [inline screens/pokedex/pokedex.gui_script:147]: Some of these functions had to be shifted up to be above the init function - they used to be called by on_input, but I switched them to be called by init, so they needed to move. I don't think I changed any of them.

**Comment by Jerakin** [comment]
Super cool! Very impressive change and a awesome improvement!

**Pull request merged**